integration/noderunner/noderunner_test.go

```diff
@@ -126,16 +126,9 @@ func tcpConnectionAvailable(addr string) bool {
 }
 
 func setupTestLog() *os.File {
-	// create a folder specific for the test
-	err := os.MkdirAll(testLogs, 0o700)
-	if err != nil {
-		panic(err)
-	}
-	timeFormatted := time.Now().Format("2006-01-02_15-04-05")
-	f, err := os.CreateTemp(testLogs, fmt.Sprintf("noderunner-%s-*.txt", timeFormatted))
-	if err != nil {
-		panic(err)
-	}
-	log.OutputToFile(f)
-	return f
+	return logutil.SetupTestLog(&logutil.TestLogCfg{
```

Comment:
I'd kill off the use of a struct - it's only three params - and avoid the wrapping of `logutil.SetupTestLog` inside `setupTestLog`, which seems unnecessary.

Reply:
I'm going to approve because it's a matter of opinion, but I disagree.

Someone will change the code - most likely not you - by adding a param that should default to `true`, and will forget to change all the usages, and a bug will appear. Declaring all the params forces you to input them all. The cost of going around putting `true` or `false` in ~10 places through the codebase is very minor IMO.

Reply:
@joeldudley yeah I hadn't considered how those default values can silently bite you that is a v good point.

But I think if it was at 4 parameters already then we wouldn't be having the debate and the person that came along to add that next param may well think "this is getting a bit hard to read, maybe it's time to wrap these up in a struct".

Anyway, thanks for approving, and we can always change it, be interesting to hear what the others think for similar examples in future (probably a resounding "I don't care" haha). Not being able to provide defaults does feel like a pretty major pain point in the language.

Reply:
Would add as well, I don't think updating things is the only cost to not using a config. The readability is potentially an even bigger factor (I know Goland can give you nice helper labels for the fields but not every reader will have those available, especially looking at git diffs and stuff).